subprojects/testing-junit-platform/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestClassProcessor.java

```diff
@@ -83,21 +86,25 @@ public void stopNow() {
         @Override
         public void execute(String testClassName) {
             Class<?> klass = loadClass(testClassName);
-            if (isTopClass(klass)) {
-                testClasses.add(klass);
+            if (isInnerClassAnnotatedWithNested(klass) || isInnerClassInsideEnclosedRunner(klass)) {
+                return;
             }
+            testClasses.add(klass);
         }
```

Comment:
I don't want to use

```
if (!isInnerClassAnnotatedWithNested(klass) && !isInnerClassInsideEnclosedRunner(klass)) {
    testClasses.add(klass);
}
```

because that's a bit harder to understand.

Reply:
A more readable pattern would be to extract the whole if condition into a method, like `shouldRun`